### DISCUSSION TITLE
No info about via relationship creation in ``$request->query`` in ``redirectAfterCreate`` method.

### DISCUSSION THREAD

**Discussion by kusab85** [laravel/nova-issues#4730] [Q&A] [answered]
- Laravel Version: 9.25.1
- Nova Version: 4.12.14 (Silver Surfer)
- PHP Version: 8.0.11
- Database Driver & Version: SqlLite 
- Operating System and Version: Windows 10 21H2
- Browser type and version: MS Edge 104.0.1293.63 64-bit and FireFox 91.12.0esr 64bit
- Reproduction Repository: https://github.com/kusab85/TestApp/tree/nova-redirectAfterCreate

### Description:

When model is created via relationship, method **``redirectAfterCreate``** **does not receive** information that resource was **created via relationship**.

Log of ``__METHOD__.': '. var_export($request->query(), true)`` in ```redirectAfterCreate``` gives:
```php
[2022-08-23 09:45:01] local.DEBUG: App\Nova\Post::redirectAfterCreate: array (
  'editing' => 'true',
  'editMode' => 'create',
) 
```

But when model is updated via relationship, method **``redirectAfterUpdate``** **does receive** information that resource was **updated via relationship**.

Log of ``__METHOD__.': '. var_export($request->query(), true)`` in ```redirectAfterUpdate``` gives:
```php
[2022-08-23 09:45:21] local.DEBUG: App\Nova\Post::redirectAfterUpdate: array (
  'viaResource' => 'users',
  'viaResourceId' => '1',
  'viaRelationship' => 'posts',
  'editing' => 'true',
  'editMode' => 'update',
)   
```

### Detailed steps to reproduce the issue on a fresh Nova installation:
- add Post model with relationship belonsTo User
- add HasMany Posts relationship to User model
- add corresponding Post Nova resource 
- add BelongsTo field to User Nova resource
- create new Post via User
- modify Post via User

**Comment by crynobone** [comment, marked as answer]
I believe those information are available from `input()` and not `query()` on the Request object.

**Reply by kusab85** [reply]
Yeah, you're right! It's in ``input``. Thx @crynobone!
